api/src/main/java/jakarta/data/repository/Select.java

```diff
@@ -25,26 +25,30 @@
 import java.lang.annotation.Target;
 
 /**
- * <p>Determines which entity attribute(s) are returned by a {@link Find}
- * operation.</p>
+ * <p>Determines the attribute or attributes of an entity which are returned
+ * by a {@linkplain Find parameter-based} query method.</p>
  *
- * <p>This annotation can be specified on a repository find method or
- * on a record component.</p>
+ * <p>This annotation may be applied to a repository method annotated
+ * {@code @Find}, or to a component of a record type.</p>
  *
- * <p>When used on a repository find method that returns an entity attribute
- * type, at most one {@code Select} annotation can annotate the method.</p>
- *
- * <p>When used on a repository find method that returns a Java record type,
- * multiple {@code Select} annotations may annotate the method, one for each
- * record component. The number and order of {@code Select} annotations must
- * match the number and order of record components.</p>
- *
- * <p>When used on a record component, the {@code Select} annotation
- * value is used for all repository methods that lack this
- * annotation and for which the record is the result type.</p>
+ * <ul>
+ * <li>At most one {@code @Select} annotation may be applied to a repository
+ * method that returns an entity attribute type.</li>
+ * <li>Multiple {@code @Select} annotations may be applied to a repository
+ * method which returns a record type, one annotation for each component of
+ * the record type. The number and order of {@code @Select} annotations must
+ * match the number and order of record components.</li>
+ * <li>At most one {@code @Select} annotation may be applied to a component
+ * of a record type. When applied to a record component, the {@code Select}
+ * annotation affects every repository method which returns the record type
+ * and which does not have a {@code @Select} annotation.</li>
+ * </ul>
  *
  * <p>This annotation must not be used in other locations.</p>
  *
+ * <p>Any repository method affected by a {@code @Select} annotation must
+ * {@linkplain Find#value explicitly specify} the queried entity type.</p>
+ *
```

Comment:
I think the point of this is to ensure users don't lose the information about which entity the query pertains to, and it is definitely good to say something about that.  However, the wording here with "explicitly specify" linking to `Find.value()` might be interpreted as ruling out two other ways of explicitly specifying: relying on the primary entity type or a JDQL `FROM` clause.  Maybe try this to avoid that confusion:

```suggestion
 * <p>Any repository method affected by a {@code @Select} annotation must
 * explicitly specify the queried entity type. This is accomplished by
 * including the {@code @Find} annotation {@link Find#value value},
 * a {@code FROM} clause (if using {@link Query @Query}), or the
 * primary entity type of the repository.</p>
 *
```

Reply:
@njr-11 I though we decided that `@Select` can't be used with `@Query`. If it can be, we need more changes than this.

Reply:
> @njr-11 I though we decided that `@Select` can't be used with `@Query`. If it can be, we need more changes than this.

There is a subtle aspect to the wording of the proposed addition here.  It says `affected by a {@code @Select} annotation` but maybe it was intended to say or mean `annotated by a {@code @Select} annotation`. If that were the case, it would not be odds with `@Query` at all. `@Select` cannot annotate a repository method that is annotated `@Query`, but `@Select` can be used on the components of a record type that is returned by a `@Query` method, which is also a way of affecting the repository method.

From `@Query`:
https://github.com/jakartaee/data/blob/527155b0ffd02cc9249200fa1be9ebe0bbbab91e/api/src/main/java/jakarta/data/repository/Query.java#L38-L42

Example in Javadoc:
https://github.com/jakartaee/data/blob/527155b0ffd02cc9249200fa1be9ebe0bbbab91e/api/src/main/java/module-info.java#L917-L928

Reply:
OK, fine, but then we need to mention in the javadoc for `@Select` that it affects `@Query` methods too, because currently it's written like `@Select` is only for use with `@Find`.

Reply:
> OK, fine, but then we need to mention in the javadoc for `@Select` that it affects `@Query` methods too, because currently it's written like `@Select` is only for use with `@Find`.

I agree.  `@Select` is currently written to avoid contradicting that usage, but it is only implicit in that `@Select` has language about "When used on a record component ..." and records can be returned by `@Query` and `@Query` says that record components can be annotated with `@Select`.  It's very indirect for a user to be able to piece that all together.  I'll see if I can propose an update to wording to be more explicit here.

Reply:
> OK, fine, but then we need to mention in the javadoc for @Select that it affects @Query methods too, because currently it's written like @Select is only for use with @Find.

I am super afraid of creating a Frankenstein at the API.
I would vote to keep the `Select`annotation only combined with `Find`.

Otherwise, we need to describe any aspect, of the combinations, such as:

```java

@Select({"id", "title", "edition"})
@Select("select title, from Book")
List<Book> find(String title);
```

Reply:
> I am super afraid of creating a Frankenstein at the API. I would vote to keep the `Select`annotation only combined with `Find`.
> 
> Otherwise, we need to describe any aspect, of the combinations, such as:
> 
> ```java
> @Select({"id", "title", "edition"})
> @Select("select title, from Book")
> List<Book> find(String title);
> ```

@otaviojava No one is proposing that we allow `@Select` and `@Query` to annotate same method (I think that is what you intended to show above rather than a `@Select` with JDQL in it).

This PR isn't changing what is allowed or adding anything new to Jakarta Data. It's only aiming to make the documentation of `@Select` more clear.  One of the patterns that it wasn't clear on is the ability to put `@Select` onto record components.  Record components can be returned by `@Query` methods as well as `@Find` methods.  Our documentation for `@Query` already states this is valid and we have other Javadoc giving [examples of this usage](https://github.com/jakartaee/data/blob/527155b0ffd02cc9249200fa1be9ebe0bbbab91e/api/src/main/java/module-info.java#L917-L928) and TCK scenarios for it.  The `@Select` documentation was already consistent with it as well, although didn't explicitly call it out.  All we are proposing to do here is have the `@Select` javadoc explicity call out what elsewhere in the documentation we already say is possible.